paasta_tools/cli/cmds/local_run.py

```diff
@@ -632,18 +632,25 @@ def configure_and_run_docker_container(
     Function prints the output of run command in stdout.
     """
 
-    soa_dir = args.yelpsoa_config_root
+    if instance is None and args.healthcheck:
+        paasta_print(
+            "With --healthcheck, --instance must be provided!",
+            file=sys.stderr,
+        )
+        sys.exit(1)
+    if instance is None and not sys.stdin.isatty():
```

Comment:
How about if 

`interactive is Tru and not tty`

I think that would achieve the same effect and be clearer, but you would need to place it further down after we've figured out if we are interactive or not.